src/incubator/TouchableOpacity2.tsx

```diff
@@ -0,0 +1,169 @@
+import React, {PropsWithChildren, useCallback, useMemo} from 'react';
+import {LayoutChangeEvent} from 'react-native';
+import Reanimated, {
+  useAnimatedGestureHandler,
+  useAnimatedStyle,
+  useSharedValue,
+  withTiming,
+  interpolate,
+  interpolateColor,
+  runOnJS
+} from 'react-native-reanimated';
+import {TapGestureHandler, LongPressGestureHandler, State} from 'react-native-gesture-handler';
+import {asBaseComponent, forwardRef, BaseComponentInjectedProps, ForwardRefInjectedProps} from '../commons/new';
+import {ViewProps} from '../components/view';
+
+export type TouchableOpacityProps = {
+  /**
+   * Background color
+   */
+  backgroundColor?: string;
+  /**
+   * Background color when actively pressing the touchable
+   */
+  feedbackColor?: string;
+  /**
+   * Opacity value when actively pressing the touchable
+   */
+  activeOpacity?: number;
+  /**
+   * Scale value when actively pressing the touchable
+   */
+  activeScale?: number;
+  /**
+   * Callback for when tapping the touchable
+   */
+  onPress?: (props: any) => void;
+  /**
+   * Callback for when long pressing the touchable
+   */
+  onLongPress?: (props: any) => void;
+  /**
+   * Pass controlled pressState to track gesture state changes
+   */
+  pressState?: State;
+  /**
+   * If true, disable all interactions for this component.
+   */
+  disabled?: boolean;
+  /**
+   * Pass custom style
+   */
+  style?: ViewProps['style'];
+  onLayout?: (event: LayoutChangeEvent) => void;
+  testID?: string;
+};
+
+type Props = PropsWithChildren<TouchableOpacityProps & BaseComponentInjectedProps & ForwardRefInjectedProps>;
+
+/**
+ * @description: a Better, enhanced TouchableOpacity component
+ * @modifiers: flex, margin, padding, background
+ * @example: https://github.com/wix/react-native-ui-lib/blob/master/demo/src/screens/incubatorScreens/TouchableOpacityScreen.js
+ */
+function TouchableOpacity(props: Props) {
+  const {
+    children,
+    modifiers,
+    style,
+    // onPress = _.noop,
+    // onLongPress,
+    disabled,
+    forwardedRef,
+    feedbackColor,
+    activeOpacity = 0.2,
+    activeScale = 1,
+    ...others
+  } = props;
+  const {borderRadius, paddings, margins, alignments, flexStyle} = modifiers;
+
+  const isActive = useSharedValue(0);
+  /* This flag is for fixing an issue with long press triggering twice
+  TODO: Consider revisiting this issue to see if it still occurs */
+  const isLongPressed = useSharedValue(false);
+
+  const backgroundColor = useMemo(() => {
+    return props.backgroundColor || modifiers.backgroundColor;
+  }, [props.backgroundColor, modifiers.backgroundColor]);
+
+  const onPress = useCallback(() => {
+    props.onPress?.(props);
+  }, [props]);
```

Comment:
Why does the onPress need all the props as an argument?
This will update the callback for every prop change

Reply:
Well technically you are right, we might consider an optimization for it. 
But see first my big comment on the props feature we pass back during press callbacks.


.....That's the reason we pass all props, and that's why we all the props are in the deps arrays. 

Anyway, I agree it's not optimal. 
Maybe we can make an assumption that the "custom" prop the user passed is a static prop that won't change, then we can ignore all the props in the deps array. It means the user will get back the initial value they passed with a custom prop. 
WDYT?

Reply:
Is that a reasonable assumption? 
Maybe the user wants to know about props change, no?

Reply:
I wanna say that yes, it's reasonable. But technically if our assumption is wrong then it might introduce a bug. 
Anyway, I have another solution. 


Not so long ago, we added a dedicate prop called `customValue` to the original TouchableOpacity component. 
See here -> https://github.com/wix/react-native-ui-lib/blob/master/src/components/touchableOpacity/index.tsx#L45

It's purpose was to be an explicit prop to pass on a custom value as I described in the previous comment. 
Also the reason we had to declare an explicit type for it is due to typings. 
When people use `<TouchableOpacity someValue={...}/>` it threw a TS error that `someValue` is not a valid prop. 
So we have `customValue` and now they can just `<TouchableOpacity customValue={...}/>`

So back to our concern, see my recent code change. 
I only condition the onPress callbacks on `props.onPress` and on `props.customValue` - which is the value that the user expect to pass on.. 

Let me know it all makes sense? (: